build-info-extractor-go/src/main/java/org/jfrog/build/extractor/go/GoDriver.java

```diff
@@ -29,13 +31,55 @@ public class GoDriver implements Serializable {
     private final CommandExecutor commandExecutor;
     private final File workingDirectory;
     private final Log logger;
+    private static final String defaultExecutablePath = "go";
 
     public GoDriver(String executablePath, Map<String, String> env, File workingDirectory, Log logger) {
-        this.commandExecutor = new CommandExecutor(StringUtils.defaultIfEmpty(executablePath, "go"), env);
+        this.commandExecutor = generateCommandExecutor(executablePath, env);
         this.workingDirectory = workingDirectory;
         this.logger = logger;
     }
 
+    /**
+     * Generate a new mutable copy of environment variables map with the Go executable directory path inserted to the beginning of the Path.
+     *
+     * @param executablePath Go executable path
+     * @param env            Environment variables map
+     * @return a new Environment variables map
+     */
+    static Map<String, String> generateWindowsEnv(String executablePath, Map<String, String> env) {
+        // If executablePath ends with "go" or "go.exe" - remove it from the directory path
+        executablePath = StringUtils.removeEnd(executablePath, ".exe");
+        executablePath = StringUtils.removeEnd(executablePath, defaultExecutablePath);
+
+        // Insert the Go executable directory path to the beginning of the Path environment variable
+        // Make sure to copy the environment variables map to avoid changing the original map or in case it is immutable
+        Map<String, String> newEnv = Maps.newHashMap(env);
+        String windowsPathEnvKey = "Path";
+        if (newEnv.containsKey(windowsPathEnvKey)) {
+            newEnv.put(windowsPathEnvKey, executablePath + File.pathSeparator + newEnv.get(windowsPathEnvKey));
+        } else {
+            newEnv.put(windowsPathEnvKey, executablePath);
+        }
+        return newEnv;
+    }
+
+    /**
+     * Create a CommandExecutor with the given executable path and environment variables.
+     *
+     * @param executablePath Go executable path
+     * @param env            Environment variables map
+     * @return CommandExecutor
+     */
+    private static CommandExecutor generateCommandExecutor(String executablePath, Map<String, String> env) {
+        if (!SystemUtils.IS_OS_WINDOWS || StringUtils.isBlank(executablePath) || StringUtils.equals(executablePath, defaultExecutablePath) || env == null) {
+            return new CommandExecutor(StringUtils.defaultIfEmpty(executablePath, defaultExecutablePath), env);
+        }
+        // Handling Windows case:
```

Comment:
But Windows cases can be in the above "if" too. Can we rearrange the code? Specifically, I'd simplify the "if" above.